dataprofiler/profilers/profile_builder.py

```diff
@@ -2112,6 +2119,65 @@ def _update_chi2(self):
 
         return chi2_mat
 
+    def _calc_nan_replication_dependent_properties(self, clean_samples):
+        """
+        Calculate metrics needed for replicating NaN values in capitalone/synthetic-data.
+
+        Metrics include class priors and class specific means of each column.
+
+        Required for running LDA based binary classifier where predicted class label indicates whether
+        a value of a column should be NaN (1) or not (0).
+
+        :param clean_samples: input cleaned dataset
+        :type clean_samples: dict
+        """
+        if not self.options.synthetic_data.replicate_nan.is_enabled:
+            return
+
+        data = pd.DataFrame(clean_samples).apply(pd.to_numeric, errors="coerce")
+
+        for col_id in range(len(self._profile)):
+            null_count = getattr(self._profile[col_id], "null_count")
+            if null_count == 0:
+                # No missing values to replicate
+                continue
+
+            # Partition data based on whether target column value is NaN or not
+            col_is_nan = data.iloc[:, col_id].isnull()
```

Comment:
could we get this info from the profile of the column already?

Reply:
This creates a boolean array needed for indexing. We do calculate the list of indices where there is a null value so I guess I could use that to create this array.

Reply:
But that would still be an O(n) operation I think (initialize an array of size `sample_size` with all values being `False` and then flip values to `True` according to list of indices where there is null)